pyEcoHAB/SetupConfig.py

```diff
@@ -0,0 +1,735 @@
+                if key.startswith("entrance") and value not in out:
+                    out.append(value)
+        return sorted(out)
+
+    def other_tunnel_antenna(self, new_antenna):
+        """
+        Return other entrance antennas to the same tunnel.
+        """
+        antenna = new_antenna
+        try:
+            tunnel_antennas = self.same_tunnel[antenna][:]
+        except KeyError:
+            return []
+        idx = tunnel_antennas.index(antenna)
+        tunnel_antennas.pop(idx)
+        return tunnel_antennas
+
+    def other_cage_antenna(self, new_antenna):
+        """
+        Return other antennas to the same cage.
+        """
+        antenna = new_antenna
+        cage_antennas = self.same_address[antenna][:]
+        idx = cage_antennas.index(antenna)
+        cage_antennas.pop(idx)
+        return cage_antennas
+
+    def next_tunnel_antennas(self, antenna):
+        """
+        Find entrance antennas to tunnels connected to the same cage
+        (specified by its entrance antenna).
+        """
+        out = []
+        same_pipe = self.same_tunnel[antenna]
+        for ant in same_pipe:
+            same_cage_a = self.other_cage_antenna(ant)
+            for a_2 in same_cage_a:
+                if a_2 in self.internal_antennas:
+                    continue
+                other_pipe = self.same_tunnel[a_2]
+                if other_pipe != same_pipe:
+                    out.extend(other_pipe)
+        return sorted(out)
+
+    def _go_two_steps(self, antenna):
+        """
+        Return antennas in tunnels that are two cages away to the right.
+        """
+        out = []
+        for a_2 in self.other_cage_antenna(antenna):
+            if a_2 in self.internal_antennas:
+                continue
+            pipes_next = self.other_tunnel_antenna(a_2)
+            for a_3 in pipes_next:
+                if a_3 in self.internal_antennas:
+                    continue
+                cage_plus_2 = self.other_cage_antenna(a_3)
+                for a_4 in cage_plus_2:
+                    if a_4 in self.internal_antennas:
+                        continue
+                    tunnel_antennas = self.same_tunnel[a_4]
+                    next_tunnel_antennas = self.next_tunnel_antennas(a_4)
+                    if antenna not in tunnel_antennas\
+                       and antenna not in next_tunnel_antennas:
+                        out += tunnel_antennas
+        return list(set(out))
+
+    def get_opposite_tunnel_dict(self):
+        """
+        Find antennas in tunnels that are two cages away
+        from current entrance antenna.
+        """
+        # distance equal two
+        all_antennas = self.entrance_antennas
+        out = {}
+        for a_1 in all_antennas:
+            out_this_antenna = self._go_two_steps(a_1)
+            other_tunnel_antennas = self.other_tunnel_antenna(a_1)
+            out_other_tunnel_antenna = []
+            for other_tunnel_antenna in other_tunnel_antennas:
```

Comment:
this looks a bit risky to me with the names other_tunnel_antenna and other_tunnel_antennas, also out_other_tunnel_antennas